src/Native/System.Native/pal_process.cpp

```diff
@@ -17,6 +17,12 @@
 #include <syslog.h>
 #include <unistd.h>
 
+#if HAVE_LIBPROC
+#include <libproc.h>
+#include <sys/proc_info.h>
+#include <sys/resource.h>
+#endif
```

Comment:
Yes, it will definitely be beneficial to switch over to sysctl in order to support FreeBSD and not have an OS X specific interface. This pack is more about shim'ing out native calls to make it easier for that to happen down the road. I'm trying to get all PInvoke code switched over to the PAL in time for a release candidate; once the PInvoke calls have been shim'd, I'd be happy to switch this over to sysctl (if it hasn't been done already) :)